app/components/ProjectRevision/ChangeReasonWidget.tsx

```diff
@@ -0,0 +1,116 @@
+import { useUpdateProjectRevision } from "mutations/ProjectRevision/updateProjectRevision";
+import { Button, Textarea } from "@button-inc/bcgov-theme";
+import { WidgetProps } from "@rjsf/core";
+import { useState } from "react";
+import { graphql, useFragment } from "react-relay";
+
+const ChangeReasonWidgetFragment = graphql`
+  fragment ChangeReasonWidget_projectRevision on ProjectRevision {
+    id
+    changeStatus
+  }
+`;
+
+// Custom widget to update the revision change reason
+const ChangeReasonWidget: React.FC<WidgetProps> = (props) => {
+  const { value, formContext, onChange } = props;
+  const projectRevision = formContext.projectRevision;
+
+  const { id, changeStatus } = useFragment(
+    ChangeReasonWidgetFragment,
+    projectRevision
+  );
+
+  const [informationalText, setInformationalText] = useState("");
+  const [initialValue, setInitialValue] = useState(value ?? "");
+
+  const [updateProjectRevision, isUpdatingProjectRevision] =
+    useUpdateProjectRevision();
+
+  const updateRevisionStatus = () => {
+    updateProjectRevision({
+      variables: {
+        input: {
+          id: id,
+          projectRevisionPatch: { changeReason: value },
+        },
+      },
+      optimisticResponse: {
+        updateProjectRevision: {
+          projectRevision: {
+            id: id,
+          },
+        },
+      },
+      onCompleted: () => {
+        setInformationalText("Updated");
+        setInitialValue(value);
+      },
+      debounceKey: id,
+    });
+  };
+
+  const changeHandler = (e) => {
+    const newValue = e.target.value;
+    // set the text next to the button based on whether the value has changed or not
+    setInformationalText(
+      initialValue !== newValue
+        ? 'To confirm your change, please click the "Update" button.'
+        : ""
+    );
+    onChange(newValue);
+  };
+
+  return (
+    <div>
+      {changeStatus === "pending" ? (
+        <div>
+          <Textarea
+            value={value}
+            size="medium"
+            resize="vertical"
+            onChange={changeHandler}
+          />
+          <div>
+            <Button
+              type="submit"
+              onClick={updateRevisionStatus}
+              style={{ marginRight: "1rem" }}
+              disabled={
+                initialValue === (value ?? "") || isUpdatingProjectRevision
+              }
+              size="small"
+            >
+              Update
+            </Button>
+          </div>
+          <small>{informationalText}</small>
```

Comment:
I think we still need the `informationalText` state since we update it once `updateProjectRevision` is completed; we change its value to `Updated` once the mutation is done.
I can make `<small>` tag optional to not render the empty tag though.